### ISSUE TITLE
Legal support in CSL and CSLm

### ISSUE THREAD

**Issue by georgd** [citation-style-language/schema#353]
_Originally posted by @bdarcus in https://github.com/citation-style-language/schema/issues/346#issuecomment-670939918_

> We maybe need to come up with a principle by which we distinguish between legal support in CSL proper, and fuller-featured legal support in Juris-M.

I think, it’s a good idea to make this explicit.

> While it may not be so clear, perhaps CSL supports legal citations in non-legal fields (generic legal support, for history, sociology, etc.), while Juris-M does law proper?

In general, I agree with you. Currently, I’m hesitating to promote Juris-M in German speaking countries because there are still pieces missing, in the schema and in the processor but this is a transitional state that I hope might end soon.

The question is: where to draw the line to decide which parts should go into CSL and which into CSLm. [side note: I have a very strong opinion about the multilingual things in CSLm to belong in vanilla CSL.]

Here are my thoughts:

# The two cases

- Full support for every aspect of legal citation — legislation, jurisdiction, executive orders etc., including parallel citation for related items, and also specific legal secondary sources like commentaries. This would be only possible with CSLm.

- Support for a basic set of features for legal citation in non-legal fields. This should be possible with CSL.

# The current situation

- The legal parts in CSL were seemingly designed with the English/American legal systems in mind. Basic needs for those might be covered. Other systems might fit or might be made to fit in the existing schema with certain necessary features missing. 
- Only a low number of legal styles are present as CSLm styles - Most legal styles are written in CSL.
- The legal styles in CSL are using many uncoordinated workarounds, so style-specific entry is very common. As an example, there are at least four Austrian legal styles, which are all incompatible one to each other, data-entry-wise, and incompatible to many non-legal styles, so somebody working in law and a second field might have troubles.

# Principles

- The complex logic needed for law-specific features is restricted to CSLm.
- Variables not needed in non-law-specific styles don’t need to be defined in CSL.
- Style specific entry should be kept at a minimum even across the CSL/CSLm border: Where CSL and CSLm share variables, a user shouldn’t have to change data-entry depending on the CSL-extension in use.
- Methods necessary in both CSLm and CSL should be implemented likewise.

# Consequences

- Legal parallel citation mechanisms, style modularisation and legal abbreviation handling belong to the realm of CSLm.
- Variables or behaviour indispensable for basic legal citation should be introduced to CSL, e.g.:  
    - https://github.com/citation-style-language/schema/issues/320 (probably requires also some wording in the CSL style repository policy about its use and non-use in CSL-styles?)
    - https://github.com/citation-style-language/schema/issues/339 (unless CSLm is going to implement it differently)
- Other variables needed for legal citations should be evaluated if they are of use in CSL, as e.g.:  
    - locator variables in https://github.com/citation-style-language/schema/issues/346
    - the `commenter` variable (cf. https://github.com/citation-style-language/schema/issues/320#issuecomment-661804113)

**Comment by bwiernik** [comment]
My opinion is that standard CSL should generally not expand its support for legal citation outside of what is already there. My sense is that the legal support there is generally suffices for non-legal researchers who need to cite an occasional legal item. 

Actual legal scholars and practitioners should use CSLm, or perhaps we at some point develop an "official" legal module to CSL. 

I don't think a halfway solution in standard CSL is useful. It doesn't actually generate correct legal citations, increases processor effort substantially, and is frankly redundant with the existing work on modular jurisdiction support developed by the Jurism/CSLm community.  

Legal citation is extremely complex, especially considering differences across jurisdictions. I don't think we should expect all citation processors to be able to handle that. 

@georgd If you find the current state of CSLm dissatisfactory, I suggest you focus on improving it. For example, if there is a jurisdiction missing from the [Legal Resource Registry](https://GitHub.com/juris-m/legal-resource-registry) that powers CSLm's cross-jurisdiction modules, contribute it to the registry. If there is something missing from the CSLm schema, raise that with @fbennett and others working in that area. Submit pull requests against the CSLm schema. 

The CSLm schema was developed with comparative international law in mind. It wasn't designed to reflect just US/UK law.

**Comment by bwiernik** [comment]
Regarding the issue of legal styles in standard CSL/style-specific data entry, I wish those styles didn't exist and were instead CSLm styles. My recommendation there is to develop CSLm versions of those styles and encourage folks to move to them.

**Comment by bwiernik** [comment]
Specifically on the issue of multi-jurisdiction support, I do not think it would be a good situation to end up with many unique CSL styles duplicating conditional jurisdiction logic and formatting. That will lead to a maintenance mess to keep the legal parts of various styles up to date and accurate. This is why @fbennett developed the modular jurisdiciton system for CSLm, so that the formatting for a jurisdiction could be programmed once and then reused in every style.

To sum up my view:
1. Parallel legal citation is outside the scope of CSL
1. Multiple-jurisdiction legal citation is outside the scope of CSL

These are the two major complexities of legal citation that I do not think should be baked into standard CSL.

I don't see any problem with adding variables like commenter or the locators in #346; those don't change processor behavior at all.

**Comment by georgd** [comment]
> My opinion is that standard CSL should generally not expand its support for legal citation outside of what is already there. My sense is that the legal support there is generally suffices for non-legal researchers who need to cite an occasional legal item.

You’re very probably right for a US researcher, but not in case of somebody in the EU.

> Actual legal scholars and practitioners should use CSLm, or perhaps we at some point develop an "official" legal module to CSL.

Agreed.

> I don't think a halfway solution in standard CSL is useful. It doesn't actually generate correct legal citations, increases processor effort substantially, and is frankly redundant with the existing work on modular jurisdiction support developed by the Jurism/CSLm community.

I don’t think so. From a lawyer’s point of view these solutions are halfway, agreed. But non-legal styles here in central Europe usually only provide for citing statutes, which is not really complex at all, citing a chapter in a book is more complicated. And correct citations thereof can be accomplished even from the lawyer’s point of view. The feature requests I posted here don’t really add to the complexity on the processor or style side, as they use methods that are in use already (simple variables, a condition that tests for equivalence against a list of strings). On the contrary, they’d take out complexity on the style side.

> Legal citation is extremely complex, especially considering differences across jurisdictions. I don't think we should expect all citation processors to be able to handle that.

Expecting this in a FOSS world is definitely not my attitude. Therefore, I think treating full legal support as an extension to CSL is a good way to go.

> @georgd If you find the current state of CSLm dissatisfactory, I suggest you focus on improving it. For example, if there is a jurisdiction missing from the [Legal Resource Registry](https://GitHub.com/juris-m/legal-resource-registry) that powers CSLm's cross-jurisdiction modules, contribute it to the registry. If there is something missing from the CSLm schema, raise that with @fbennett and others working in that area. Submit pull requests against the CSLm schema.

I’ve been contributing to the LRR already and I’ve been in contact with Frank about these issues for over a year. But that’s a different story.

> The CSLm schema was developed with comparative international law in mind. It wasn't designed to reflect just US/UK law.

I didn’t say that about CSLm but about vanilla CSL — I’m sorry if it was unclear.

**Comment by georgd** [comment]
> Specifically on the issue of multi-jurisdiction support, I do not think it would be a good situation to end up with many unique CSL styles duplicating conditional jurisdiction logic and formatting. That will lead to a maintenance mess to keep the legal parts of various styles up to date and accurate. This is why @fbennett developed the modular jurisdiciton system for CSLm, so that the formatting for a jurisdiction could be programmed once and then reused in every style.
> 
> To sum up my view:
> 
> 1. Parallel legal citation is outside the scope of CSL
> 2. Multiple-jurisdiction legal citation is outside the scope of CSL
> 
> These are the two major complexities of legal citation that I do not think should be baked into standard CSL. "Basic legal citation" should be understood to mean "a researcher in a non-legal field cites a small number of legal items from a single jurisdiction".

I’m sorry, but that’s not the reality. EU statutes are an integral part of every EU member’s legislation. Thus, citing Austrian law may mean citing EU statutes. EU law may be directly valid without the necessity of the member state to implement it in its own law. Thus, even if you only cite Austrian law, you will end up citing EU law as well.

I understand your point and I support your view to not duplicate the modular jurisdiction system in a huge number of monstrous monolithic vanilla CSL styles. That’s why I proposed to clarify this in the repository policy.

If you find a better solution to the EU case that wouldn’t open the door to this issue, I‘m fine with it as well. Just, the current situation is all but optimal.

**Comment by bdarcus** [comment]
> I’m sorry, but that’s not the reality. EU statutes are an integral part of every EU member’s legislation. Thus, citing Austrian law may mean citing EU statutes. EU law may be directly valid without the necessity of the member state to implement it in its own law. Thus, even if you only cite Austrian law, you will end up citing EU law as well.

And where this hits CSL is different jurisdictions have different styling expectations in many styles?

**Comment by georgd** [comment]
> > I’m sorry, but that’s not the reality. EU statutes are an integral part of every EU member’s legislation. Thus, citing Austrian law may mean citing EU statutes. EU law may be directly valid without the necessity of the member state to implement it in its own law. Thus, even if you only cite Austrian law, you will end up citing EU law as well.
> 
> And where this hits CSL is different jurisdictions have different styling expectations in many styles?

That’s right. That’s a thing, that can’t be accomplished in the big internationally used styles without the modular system of CSLm. As they all originate in Northern America, the legal citations are fixed to something akin to US traditions (so, in a sense there’s always jurisdiction-specific formatting). I don’t expect this in the monolithic styles. However, there are lots of styles with regionally restricted user communities that could very well profit from such condition.

**Comment by denismaier** [comment]
Perhaps there's a third way to approach this problem: Leave support for law out of CSL completely, but add ways to enhance vanilla CSL with additional features or sets of features.  Other disciplines could have their own enhancements, e.g. classics or so. Or there could be a multilingual module, or whatever... Of course, those add-ons should be compatible with each other.

**Comment by bdarcus** [comment]
When ideas like this have come up, I've wondered: how would this idea of
"add-ons" or "extensions" actually work? Do we have any idea?

>

**Comment by denismaier** [comment]
> * Parallel legal citation is outside the scope of CSL
> 
> * Multiple-jurisdiction legal citation is outside the scope of CSL

Regarding the apparent complexity of these two features, I tend to agree that proper support for parallel legal citations may indeed be too complex, but I'm not so sure regarding multiple jurisdictions. Doesn't that mean that processors choose a different styles based on the item type of a given item and the value in a certain field (in that case CSLm's jurisdiction`)? 
I'm probably missing something, but that doesn't seem so utterly complex.

**Comment by georgd** [comment]
> > * Parallel legal citation is outside the scope of CSL
> > * Multiple-jurisdiction legal citation is outside the scope of CSL
> 
> Regarding the apparent complexity of these two features, I tend to agree that proper support for parallel legal citations may indeed be too complex, but I'm not so sure regarding multiple jurisdictions. Doesn't that mean that processors choose a different styles based on the item type of a given item and the value in a certain field (in that case CSLm's jurisdiction`)?
> I'm probably missing something, but that doesn't seem so utterly complex.

In fact, I do see the complexity in the latter if you modularise the styles. If you stay with monolithic style files it’s less a question of complexity but rather of messiness that makes the styles unmaintainable if more than a couple of jurisdictions is implemented, IMO.

**Comment by bdarcus** [comment]
What kind of differences are there across jurisdictions?

Minor or major?

What do they look like?

>

**Comment by denismaier** [comment]
> When ideas like this have come up, I've wondered: how would this idea of "add-ons" or "extensions" actually work? Do we have any idea?

Not really... ;-) @cormacrelf has had some ideas in the past. Maybe he want's to chime in... Also cc @fbennett, @PaulStanley, @jgm, @andras-simonyi

Or maybe: for multilingual support citeproc-js currently accepts some processor directives:

```
>>===== LANGPARAMS =====>>
{
    "institutions": [
        "orig"
    ],
    "persons": [
        "orig"
    ],
    "titles": [
        "orig",
        "translat"
    ],
    "journals": [
        "orig"
    ],
    "places": [
        "orig"
    ],
    "publishers": [
        "orig"
    ]
}
<<===== LANGPARAMS =====<<
```

Different areas might allow for similar things. You could save this in config file pass that somehow to the processor.

Or, for legal referencing with multiple jurisdictions,

In a pandoc document, I imagine you could do something like this:

```
csl: 
  style: chicago-note-bibliography.csl
  features: x, y, z
  jurisdiction-specific-citations: yes
  multilingual-configuration:
    institutions: orig
    persons: orig
    titles:
    - orig
    - translation	
```

**Comment by denismaier** [comment]
> In fact, I do see the complexity in the latter if you modularise the styles.

Really, isn't that just another conditional? Or another pre-processing step? Perhaps @fbennett could chime in, so we could properly assess what really would be needed here...

**Comment by georgd** [comment]
> What kind of differences are there across jurisdictions? Minor or major? What do they look like?
> […](#)

- Austrian statute:  
  - in Bibliography: IVF-Fonds-Gesetz, BGBl I 1999/180 idF BGBl I 2004/42.

        [title], [gazette] [year]/[number] idF [amended gazette] [amended year]/[amended number].

  - short citation with locator: §7 IVF-Fonds-Gesetz 1999
- EU regulation:  
  - in Bibliography: Verordnung 2015/478 des Europäischen Parlaments und des Rates vom 11. März 2015 über eine gemeinsame Einfuhrregelung, ABl 2015 L 83/16 [Gemeinsame Einfuhrregelungs-VO].

        [type of law| [year]/[law number] [authorities] [date] [matter], [gazette] [year] [series]
        [volume]/[starting page] [[short title]].

  - short citation with locator: Art 2 Gemeinsame Einfuhrregelungs-VO  
    or, without assigning a short title: Art 2 VO 2015/478

[If you are wondering: The narrative form of the EU regulations are challenging. In languages like German or slavonic languages with case inflections you have to put at least the components `authorities` up to `matter` in the title field, if not also the first three, in order to have a nice view Zotero’s title column.]

**Comment by the-solipsist** [comment]
One note I would add: some popular style manuals provide citation guidance for some level of legal citation.  It would be useful for the style manual collective hive mind to collate examples, and see what makes sense to include in CSL rather than CSL-M.

One shortcoming, for instance, is that the current CSL doesn't distinguish between ordinary statutes and constitutions (both fall under "legislation"), whereas even popular styles such as [CMS, APA, and MLA do require a separate style for constitutions](https://penandthepad.com/cite-us-constitution-21426.html) (by notably, omitting the year field).

Having slightly more robust support for legal citations (at least to the extent that citation styles that aren't law-specific support them), would certainly be helpful.

Further, I really like @denismaier's proposal.  The main reason I see for CSL-M being separate is that its complexity isn't desired in regular CSL and that some modifications (rather than just additions) are required for it.  But the main part of that complexity, from what I understand, isn't mostly in the schema itself, but in (a) the specific styles that choose to use the extended schema and choose to implement complex conditional logic with them; and (b) the reference manager's UI and processing to manage this complex logic.  (Please let me know if my understanding is incorrect, as it well may be.)

The practical problem I see is that the extended schema cannot be used even if specific styles do not use complex conditional logic, they are still deprived of the additional item types, elements, variables, attributes, and locator terms.  Further, any use of these extensions will fail CSL schema validation.

It would be useful to have a system where:
1. The citation software chooses whether to use a module or not depending on the style used.  (Thus, citing laws in APA/MLA won't require use of the module, whereas citing laws using OSCOLA/Bluebook will.)
2. The CSL schema validator can validate the styles that use extensions provided by these modules.

(If the above doesn't make sense, then please feel free to ignore it.  I'm a neophyte in the world of CSL, and may have misunderstood how things work.  But I felt I could try to contribute to the extent that I did feel I understood.)
